apps/condo/domains/acquiring/schema/Payment.js

```diff
@@ -494,11 +495,59 @@ const Payment = new GQLListSchema('Payment', {
             }
         },
         afterChange: async ({ context, operation, existingItem, updatedItem }) => {
+            // Trigger webhook task on status change FIRST - this must always happen
+            // regardless of any subsequent operations that might fail
+            const previousStatus = get(existingItem, 'status')
+            const newStatus = get(updatedItem, 'status')
+            const statusChanged = operation === 'update' && previousStatus !== newStatus
+
+            // Use a small closure to cache invoice loading within this hook execution.
+            // This avoids multiple getById('Invoice', ...) calls when both webhook enqueueing
+            // logic and invoice status update logic need the invoice.
+            let invoice = null
+            let invoiceFetched = false
+            const getInvoice = async () => {
```

Comment:
Please define it globally ... you don't need to redefine it every request

Reply:
This is closure. Invoices are different for different payments. This is just a local cache to prevent duplication of getById() call